Zend/zend_API.c

```diff
@@ -1846,6 +1846,24 @@ ZEND_API zend_result object_init_ex(zval *arg, zend_class_entry *class_type) /*
 }
 /* }}} */
 
+ZEND_API zend_result object_init_with_constructor(zval *arg, zend_class_entry *class_type, uint32_t param_count, zval *params) /* {{{ */
+{
+	zend_result status = _object_and_properties_init(arg, class_type, NULL);
+	if (UNEXPECTED(status == FAILURE)) {
+		return FAILURE;
+	}
+	/* A constructor does not return a value */
+	zend_call_known_instance_method(
+		class_type->constructor,
```

Comment:
But here I'd probably derive the constructor from the `arg`: `Z_OBJCE_P(arg)->constructor`.

Reply:
I find it clear as-is